frontend/src/metabase/containers/UndoListing.jsx

```diff
@@ -62,49 +54,57 @@ UndoToast.propTypes = {
 };
 
 function UndoToast({ undo, onUndo, onDismiss }) {
-  const style = undo.toastColor
-    ? { backgroundColor: color(undo.toastColor) }
-    : undefined;
   return (
-    <ToastCard dark data-testid="toast-undo" style={style}>
-      <CardContent>
-        <CardContentSide>
-          <CardIcon name={undo.icon || "check"} color="white" />
-          {renderMessage(undo)}
-        </CardContentSide>
-        <CardContentSide>
-          {undo.actions?.length > 0 && (
-            <UndoButton onClick={onUndo}>{t`Undo`}</UndoButton>
-          )}
-          <DismissIcon name="close" onClick={onDismiss} />
-        </CardContentSide>
-      </CardContent>
-    </ToastCard>
+    <Motion
+      defaultStyle={{ opacity: 0, translateY: 100 }}
+      style={{ opacity: spring(1), translateY: spring(0) }}
+    >
+      {({ translateY }) => (
+        <ToastCard
+          dark
+          data-testid="toast-undo"
+          style={{
+            ...{ transform: `translateY(${translateY}px)` },
+            ...(undo.toastColor
+              ? { backgroundColor: color(undo.toastColor) }
+              : {}),
+          }}
+        >
+          <CardContent>
+            <CardContentSide>
+              <CardIcon name={undo.icon || "check"} color="white" />
+              {renderMessage(undo)}
+            </CardContentSide>
+            <CardContentSide>
+              {undo.actions?.length > 0 && (
+                <UndoButton onClick={onUndo}>{t`Undo`}</UndoButton>
+              )}
+              <DismissIcon name="close" onClick={onDismiss} />
+            </CardContentSide>
+          </CardContent>
+        </ToastCard>
+      )}
+    </Motion>
   );
 }
 
-UndoListing.propTypes = {
-  undos: PropTypes.array.isRequired,
-  performUndo: PropTypes.func.isRequired,
-  dismissUndo: PropTypes.func.isRequired,
-};
+function UndoListingInner() {
+  const dispatch = useDispatch();
+  const undos = useSelector(state => state.undo);
```

Comment:
I deleted the `getUndo` selected and `undo.js` since it seemed overkill to have an entire separate selector and file that was located far from the single component that uses it.